### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by rraammiinn** [vlang/v#17241] [Questions and Answers]
is it faster than go ?

**Comment by Wajinn** [comment]
I'm going go ahead and respond to this, as if a legit question.  But, not putting real titles on questions does a disservice to other V users, because they won't be able to more easily search on the question.  In my opinion, the question and title comes across as a bit suspicious, so hope there is no mischievous intent.  Though I do realize that it can also come from a place of being inexperienced or unfamiliar, and if so, would like to be helpful.  

Both V and Go are compiled languages.  Consequently and relatively, they are going to be among the faster programming languages.  When dealing with compiled languages, many different factors have to be taken into consideration.  The quality of code written, algorithms used, test conditions, type of test, being optimized for the tests in question, people of a particular language focusing on and  submitting lots of examples for a test, etc...  So among the compiled languages, we can see lots of jocking and changes of rankings concerning speed or performance, depending on which month, which site, or the tests used.

They are both fast, and any slight differences, are usually inconsequential to small for the majority of users and in most use cases.  The next "elephant in the room", deals with GC.  Here too, the difference in performance versus memory safety and convenience, means that using a compiled language with a GC is of little to no consequence in the vast majority of situations.  Also, the greater memory safety and convenience that a GC can provide, should not be casually disregarded in a blind quest to squeeze out an "extra ounce" of speed without some very legitimate reasons or the person knowing exactly what they are doing.

V does have an advantage over Go, where its GC can be turned off.  That's also no slap or disrespect of Go, as you can tweak their GC in various ways, in attempts to squeeze some extra out.  V users can do manual memory management to their heart's content and tweak it to higher levels of performance.  After which, V can also be directly transpiled to C (.c as oppose to .v).  Which relates to your other non-titled thread (which included V compared to C in terms of speed).  Of course, this is my perspective about the differences, others may think differently.

**Comment by medvednikov** [comment]
v -prod is significantly faster than Go for the same code, since it uses `clang/gcc -O2`. C's optimization is much better and mature than Go's, although they are catching up.

Dramatically slower to compile though, at least 15x slower.

**Reply by rraammiinn** [reply]
I don't care about compile time , I'm only interested in runtime .
I am very interested in vlang specially it's concurrency , pattern matching and error handling but I'm wondered if it's worthy to learn it while there isn't any youtube tutorial on it .
It's very nicer than go but less mature .
some people say that it's as fast as c and others say it's even slower than go but even if it's something between go and c it will be a great language .

**Reply by JalonSolov** [reply]
Whether or not to learn it now is entirely your decision.  V is stable and functional enough to get a lot of stuff done, but there may still be some changes before the 1.0 release.  Most of those will be easily taken care of just by running `v fmt` on your code.

**Reply by MatejMagat305** [reply]
> I don't care about compile time , I'm only interested in runtime . I am very interested in vlang specially it's concurrency , pattern matching and error handling but I'm wondered if it's worthy to learn it while there isn't any youtube tutorial on it . It's very nicer than go but less mature . some people say that it's as fast as c and others say it's even slower than go but even if it's something between go and c it will be a great language .

Vlang is young and imature, but now is an exciting time when an ordinary programmer (like myself) can contribute to its development and possibly add features that in mature lang. is not possible add. I am new to Vlang and have contributed little thing(and maybe I'm litle bit ennoying for main contributors), but let me share my opinion: 
Vlang will be as easy to learn as Python and as strong and fast as C. Currently, the GC slightly limits parallelization, but compared to Go, Vlang already has much more options, for example, if you don't want GC and you are sure you don't need it, you can turn it off (and not include it in the binary), if you want a different thread size, you can compile it with a different stack, compared to Go, there is no overhead with C libraries, there is a very smart compiler... the strength of Vlang is in its simplicity and yet great ability to choose from options with comparable power to C... is great language

**Comment by Wajinn** [comment]
> ...it's worthy to learn it while there isn't any youtube tutorial on it .

There are Vlang videos on YouTube.  Might be the search terms used, that resulted in not finding them.  Of course even more, would be even better.

[An introduction to V - the vlang](https://youtu.be/ZO3vlwqvTrc)
[Coderlyfe V programming playlist](https://www.youtube.com/playlist?list=PLvLz9K34Y26baplmBi-3oJNA7gzb5Nqui)
[TutLinks V Lang Programming playlist](https://www.youtube.com/watch?v=BVCuZ7z7GMY&list=PLEPMhdsq-gNpFr40A-ZnX-Hu9l-Sp5Oc_)
[The V Programming Language videos](https://www.youtube.com/@VLang/videos)

**Comment by rraammiinn** [comment]
I wished it can be used for embedded development and even if not it's still great for application development .
I think rust is very complicated for application development and not suitable for microcontrollers .
zig can be very good for embedded but too difficult for application development .

**Reply by MatejMagat305** [reply]
you can put flag `-freestanding`, `-gc none` and `-no_bounds_checking` (maybe `-autofree` or `-prealoc`) and it is embeded suport, please you should try `v help`, `v help build` and `v help build-c`
